docs/bbe-guidelines/bbe-guidelines.md

```diff
@@ -0,0 +1,174 @@
+*   **No restriction for the max chars in comment lines as they will go to RHS side and wraps automatically. But since users can refer the code in GitHub, it is better if we can have the same char limit as the code line (i.e. 80) as it increases the readability of the code file.**
+
+Ex: Comments are significantly longer than the code line, which is not readable. 
+
+
+
+<p id="gdcalert4" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/Guidelines-in3.png). Store image on your image server and adjust path/filename if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert5">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/Guidelines-in3.png "image_tooltip")
+
+
+
+
+*   **All keywords and any other word which needs to be highlighted should be used with backquotes. (ex:  `xml`).  Don’t use a single quote as it won’t get highlighted. **
+
+Ex:
+
+Code:
+
+
+
+<p id="gdcalert5" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/Guidelines-in4.png). Store image on your image server and adjust path/filename if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert6">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/Guidelines-in4.png "image_tooltip")
+
+
+b.io:
+
+
+
+<p id="gdcalert6" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/Guidelines-in5.png). Store image on your image server and adjust path/filename if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert7">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/Guidelines-in5.png "image_tooltip")
+
+
+
+
+*   **We use a common pattern in `.out` files. Use the following format and customize only when necessary (ex: When needed to add more command line args etc)**
+
+<span style="text-decoration:underline;">For an example with main:</span>
+
+# To run this sample, navigate to the directory that contains the
+
+# `.bal` file and issue the `ballerina run` command.
+
+$ ballerina run <sample_file_name>.bal
+
+<span style="text-decoration:underline;">For an example with a service:</span>
+
+# To start the service, navigate to the directory that contains the
+
+# `.bal` file and issue the `ballerina run` command.
+
+$ ballerina run hello_world_service.bal
+
+
+
+*   **As a practice, we use `<code>ballerina/io`</code> methods in main examples and `<code>ballerina/log</code>` in services examples. Don’t use both `<code>io:println</code>` and `<code>log:printInfo</code>` in the same sample.</strong>
+*   <strong>Remove unused imports in `.bal` files.</strong>
+*   <strong>Format the ballerina source using an IDE Plugin. </strong>
+*   <strong>Always think about the length of the sample. A good example should be short.</strong>
```

Comment:
And it should be self-contained as much as possible. i.e., should avoid depending on concepts other than the one being illustrated  as much as possible.

Reply:
Added this as follows under "General best practices".

"Make the examples self-contained (i.e., avoid depending on concepts other than the one being illustrated) as much as possible."